### PR TITLE
Соловьев Данила. Задача 1. Вариант 15. Технология SEQ. Сортировка Шелла с простым слиянием.

### PR DESCRIPTION
**Pull request by DSolo03**
Сортировка Шелла с простым слиянием является модифицированной версией сортировки вставками, где элементы массива сортируются с использованием различных интервалов между элементами.
Начальный интервал, `gap` - изначально равен длины массива.
Сортируются все подмассивы используя сортировку вставками, индексы подмассивов разделены интервалом `gap`.
После этого, интервал уменьшается в два раза и происходит повторная сортировка.
Алгоритм продолжает работу, пока длина интервала не будет равна 1, при длине интервала 1 массив является отсортированным.

### CODE REVIEW

File: tasks/seq/solovyev_d_shell_sort_simple/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  for (size_t i = 0; i < data.size(); i++) {
+    if (data[i] < last) {
+      return false;
+    }
+    last = data[i];
+  }
+  return true;
+}
+}  // namespace
+
```
Comment (aobolensk) [inline tasks/seq/solovyev_d_shell_sort_simple/func_tests/main.cpp:24]: Please, add random tests as well

**Review by sudo-technoir — CHANGES_REQUESTED** [review]

**Comment by Save-Yourself** [comment]
you probably should to add tests with an empty array and a test with negative numbers.

**Review by Save-Yourself — APPROVED** [review]

File: tasks/seq/solovyev_d_shell_sort_simple/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#include "core/task/include/task.hpp"
+#include "seq/solovyev_d_shell_sort_simple/include/ops_seq.hpp"
+
+namespace {
+std::vector<int> GetRandomVector(int sz) {
+  std::random_device dev;
+  std::mt19937 gen(dev());
+  std::vector<int> vec(sz);
+  for (int i = 0; i < sz; i++) {
+    vec[i] = (int)(gen() % 100);
```
Comment (aobolensk) [inline tasks/seq/solovyev_d_shell_sort_simple/func_tests/main.cpp:19]: Please, consider adding tests for negative numbers

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]

File: tasks/seq/solovyev_d_shell_sort_simple/func_tests/main.cpp [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+namespace {
+std::vector<int> GetRandomVector(int sz) {
+  std::random_device dev;
+  std::mt19937 gen(dev());
+  std::vector<int> vec(sz);
+  for (int i = 0; i < sz; i++) {
+    int negative_factor = (int)(gen() % 100);
+    vec[i] = (int)(gen() % 100);
+    if (negative_factor % 2 == 0) {
+      vec[i] = vec[i] * -1;
```
Comment (allnes) [inline tasks/seq/solovyev_d_shell_sort_simple/func_tests/main.cpp:22]: please simplify with help range [a, b] - `(gen() % (b - a)) + a`

**Review by allnes — APPROVED** [review]

File: tasks/seq/solovyev_d_shell_sort_simple/src/ops_seq.cpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#include "seq/solovyev_d_shell_sort_simple/include/ops_seq.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <vector>
+
+bool solovyev_d_shell_sort_simple_seq::TaskSequential::PreProcessingImpl() {
+  unsigned int input_size = task_data->inputs_count[0];
```
Comment (sudo-technoir) [inline tasks/seq/solovyev_d_shell_sort_simple/src/ops_seq.cpp:8]: you should use size_t
